src/frontend/screens/JoinProject.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from "react";
+import { Alert, StyleSheet, View } from "react-native";
+import { FormattedMessage, defineMessages } from "react-intl";
+import QRCode from "react-native-qrcode-svg";
+import { getUniqueId } from "react-native-device-info";
+
+import useWifiStatus from "../hooks/useWifiStatus";
+import { MEDIUM_BLUE, WHITE } from "../lib/styles";
+import { BackIcon } from "../sharedComponents/icons";
+import Button from "../sharedComponents/Button";
+import HeaderTitle from "../sharedComponents/HeaderTitle";
+import IconButton from "../sharedComponents/IconButton";
+import Text from "../sharedComponents/Text";
+import WifiBar from "../sharedComponents/WifiBar";
+
+const m = defineMessages({
+  title: {
+    id: "screens.JoinProject.title",
+    defaultMessage: "Join a Project",
+  },
+  instructionsTitle: {
+    id: "screens.JoinProject.instructionsTitle",
+    defaultMessage: "Instructions",
+  },
+  instructionsDescription: {
+    id: "screens.JoinProject.instructionsDescription",
+    defaultMessage: "Show this QR code to your Project Admin",
+  },
+  sendJoinRequest: {
+    id: "screens.JoinProject.sendJoinRequest",
+    defaultMessage: "Send Join Request instead",
+  },
+});
+
+const JoinProject = () => {
+  const { ssid } = useWifiStatus();
+
+  const deviceName: string =
+    "Android " + getUniqueId().slice(0, 4).toUpperCase();
+
+  return (
+    <View style={styles.pageContainer}>
+      {/* TODO: Probably want to do something smarter here if no ssid is detected */}
+      {ssid && <WifiBar deviceName={deviceName} ssid={ssid} />}
+      <View style={styles.container}>
+        <View>
+          <View style={styles.qrCodeContainer}>
+            <QRCode value="https://digital-democracy.org" size={250} />
```

Comment:
Just a placeholder. Need to figure out what to encode here

Reply:
based on discussion, seems like we want to encode public key, IP address, and port. @gmaclennan any thoughts on what to use for encoding these (not familiar with best practices if any) and also, may need to figure out how to get these details in the first place

Reply:
> based on discussion, seems like we want to encode public key, IP address, and port. 

Yes, now I'm also thinking maybe device name and/or device ID so that we can show those on the admin side when they scan the join request?

> @gmaclennan any thoughts on what to use for encoding these (not familiar with best practices if any)

The easiest option is to encode this as a JSON e.g.

```json
{
   "publicKey": "dadceeb7c5d377dfdcc0b814acb336c0cf484a02ac5a87d8408e05eddd56d7dc",
   "host": "10.532.432.1:5998",
}
```

_however_ it depends on how much "space" we have in the QR code. The above JSON is 107 characters (it's a 256-bit key encoded in hex). If space is of essence we could encode this to binary: 107 characters is 107 bytes _if_ we use only UTF-8 characters. In binary we have a 256-bit key = 32 bytes, plus 4 bytes for the IP, and 2 bytes for the port, so if we encode it as a very simple concatenated buffer of publicKey | IP | port then that's only 38 bytes. But there is a lot less flexibility there, and more work if we want to add custom length fields. We could also use protobuf to encode it. Again, it's more overhead. If the JSON encoding doesn't result in that large a QR code then we could just use that. Simple and easy to understand.

> and also, may need to figure out how to get these details in the first place

This will rely on backend Mapeo Core methods that have not yet been written.

Reply:
> Yes, now I'm also thinking maybe device name and/or device ID so that we can show those on the admin side when they scan the join request?

ah yes definitely need that too

thanks for explaining the options! my initial thought is to resort to the JSON format in that case

> This will rely on backend Mapeo Core methods that have not yet been written.

right ok then i won't consider that as blocking for this PR for the time being